schema_editor/app/scripts/json-editor/json-editor-directive.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Directive wrapper for the JSONSchema js plugin:
+ *     https://github.com/jdorn/json-editor
+ *
+ * @param {string} editorId A string identifier that can be used to retrieve the editor object
+ *                          from the associated Editor service
+ * @param {object} options A JSONEditor options object, passed directly to the constructor
+ *                         default: {}
+ * @param {function} onDataChange A function on the wrapper scope to be called when the editor
+ *                                data changes, passes new editor data as first argument.
+ *
+ * Events:
+ *      json-editor:ready
+ *          @param {JSONEditor} editor The editor object that is now ready
+ */
+(function () {
+    'use strict';
+
+    var defaults = {};
+
+    function JsonEditor() {
+
+        var editor = null;
+
+        var module = {
+            restrict: 'E',
+            scope: {
+                editorId: '@',
+                options: '=',
+                onDataChange: '&'
+            },
+            link: link
+        };
+        return module;
+
+        function link(scope, element) {
+            var htmlElement = element[0];
+            var changeRef = null;
+
+            // Every time the editor options change, we need to destroy and recreate
+            // the form, preserving the user-entered data in the form
+            scope.$watch('options', function (newValue) {
+                if (!(newValue && newValue.schema)) {
+                    return;
+                }
+                var options = angular.extend({}, defaults, scope.options);
+                delete options.startval;
```

Comment:
You're right, that's left over from when I was testing different ways to reload the user-entered values in the form when it gets destroyed + recreated. 

I'll remove and tweak this a bit, because as it stands now, the startval property on options will never be applied.